corehq/apps/users/migrations/0072_remove_location_fields_in_user_data.py

```diff
@@ -0,0 +1,63 @@
+from django.db import migrations
+
+from couchdbkit import ResourceNotFound
+
+from corehq.apps.users.util import user_location_data
+from corehq.apps.users.models import CommCareUser
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+@skip_on_fresh_install
+def remove_keys_from_data(apps, schema_editor):
+    SQLUserData = apps.get_model('users', 'SQLUserData')
+
+    for user_data in SQLUserData.objects.all():
+        data = user_data.data
+        keys_to_remove = ['commcare_location_id', 'commcare_location_ids', 'commcare_primary_case_sharing_id']
+
+        modified = False
+        for key in keys_to_remove:
+            if key in data:
+                del data[key]
+                modified = True
+
+        if modified:
+            user_data.data = data
+            user_data.save()
+
+
+def revert_keys_in_data(apps, schema_editor):
+    SQLUserData = apps.get_model('users', 'SQLUserData')
+    CommCareUserCouch = CommCareUser.get_db()
+
+    for user_data in SQLUserData.objects.all():
+        user_id = user_data.user_id
+
+        try:
+            commcare_user = CommCareUserCouch.get(user_id)
+        except ResourceNotFound:
```

Comment:
When I use `SQLUserData`'s `user_id` to query `Couch` doc, I can get `ResourceNotFound: deleted`. Does it mean that user has been deleted? And should we delete the related `SQLUserData` object as well?

Reply:
I remember deletions being a little weird when working on this.  `SQLUserData` is tied to the Django `User` for a given web user, and should be deleted automatically when that is deleted.  This sounds like a couch user was deleted, but not the corresponding Django user (and user data).  Not sure what could cause that.